### ISSUE TITLE
Simple Payments Block: Sometimes block is not rendering on frontend

### ISSUE THREAD

**Issue by brbrr** [Automattic/wp-calypso#28800]
#### Steps to reproduce
1. Starting at URL: /wp-admin/post-new.php
2. Add SP block
3. Fill in SP block details and publish block without un-focusing block
4. Check your post on frontend - SP block will be missing

The problem: At the moment of publishing - SP block did not have it's ID and shortcode created, which means post published without SP shortcode.

#### What I expected

SP block will be rendered on frontend no matter how fast or tricky my flow would be

#### What happened instead

SP block wasn't populated with shortcode/ID due to how block request this data 

#### Screenshot / Video

![screencast 2018-11-22 17-08-45](https://user-images.githubusercontent.com/5654161/48911212-e5957c00-ee7a-11e8-8f7b-3eed3e2169e8.gif)

**Comment by sirreal** [comment]
This is a race. Sort of. With a big handicap for the one we need to win.

TL;DR - we need to be eager about getting an ID so we can serialize something into the post.

- The Simple payment product is a post. It's a post within a post.
- Saving (serializing the simple payment into the post) depends on having a post id for the simple payment.
- We save the product post when we save the post it's in.
- When we first save the post, the new simple payment doesn't yet have its ID. The serialized attributes will be saved into the post content: `<!-- wp:jetpack/simple-payments {"title":"buy me" } /-->`
- When the Simple Payment block detect the save event, it attempts to save.
- If successful, the Simple Payment gets an ID, it now writes the shortcode into the post content: 
  ```html
  <!-- wp:jetpack/simple-payments {"paymentId":123} /-->
  [simple-payments id="123"]
  <!-- /wp:jetpack/simple-payments -->
  ```
- Now we have a shortcode in the post (needed to render), but we haven't saved it!
- Once we do have an ID, we don't care about which post saves first or what order.

**Comment by simison** [comment]
We could try to check from the editor if the Post is already a `draft` or only`autodraft`. Autodrafts will not show up in "drafts" list and I believe are also cleaned up by wp-cron.

https://codex.wordpress.org/Post_Status
